### PR TITLE
Add Grafana Oncall module and terragrunt config.

### PR DESCRIPTION
**Pull request by ce0la**
## Summary by CodeRabbit

- **New Features**
  - Introduced on-call scheduling with weekly shifts and calendar integration.
  - Added data retrieval configurations for users, Slack channels, and teams.
  - Implemented alert integration with routes, escalation chains, and Slack notifications.

- **Documentation**
  - Provided descriptions for new variables related to on-call access, team details, and scheduling.

- **Chores**
  - Established local variables and environment settings for Grafana on-call configuration.

### CODE REVIEW

File: infra/modules/grafana-oncall/shift.tf [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  time_zone  = "UTC"
+  duration   = 60 * 60 * 24 // 24 hours
+  frequency  = "weekly"
+  interval   = 1
+  by_day     = ["MO", "TU", "WE", "TH", "FR", "SA", "SU"]
+  week_start = "MO"
+
+  rolling_users = [for username, user in data.grafana_oncall_user.users : [user.id]]
+  start_rotation_from_user_index = 1
+}
```
[the remark this answers is not in the document]
Comment (tshaynik) [reply infra/modules/grafana-oncall/shift.tf:24]: CodeRabbit is wrong here. From the docs:

> [rolling_users](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/oncall_on_call_shift#rolling_users) (List of Set of String) The list of lists with on-call users (for rolling_users event type)

**Review by tshaynik — APPROVED** [review]
LGTM. Presumably we'll have to discuss exactly how the team does the OnCall rotation in the team meeting, and having a simple rolling rotation set up like this is a good starting point for that discussion.

File: infra/modules/grafana-oncall/shift.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
+resource "grafana_oncall_on_call_shift" "week_shift" {
```
Comment (shlevy) [inline infra/modules/grafana-oncall/shift.tf:9]: Would be good to have a comment explaining at a high level what the rotation is.

Comment (ce0la) [reply infra/modules/grafana-oncall/shift.tf:9]: Done.

**Pull request merged**